File: apps/mobile_flutter/lib/core/config/app_config.dart
class AppConfig {
  const AppConfig({
    required this.apiBaseUrl,
    required this.environmentName,
  });

  factory AppConfig.fromEnvironment() {
    const apiBaseUrl = String.fromEnvironment(
      'BHASHAVAANI_API_URL',
      defaultValue: 'http://localhost:6001',
    );
    const environmentName = String.fromEnvironment(
      'BHASHAVAANI_ENV',
      defaultValue: 'local',
    );

    return const AppConfig(
      apiBaseUrl: apiBaseUrl,
      environmentName: environmentName,
    );
  }

  final String apiBaseUrl;
  final String environmentName;

  bool get isRemote => environmentName != 'local';
}

File: apps/web_pwa/src/core/config/appConfig.ts
// Mirrors apps/mobile_flutter/lib/core/config/app_config.dart.
// Vite exposes build-time env vars prefixed with VITE_ via import.meta.env;
// set VITE_BHASHAVAANI_API_URL / VITE_BHASHAVAANI_ENV in a .env file to
// override for a given build, same idea as Flutter's --dart-define.
export interface AppConfig {
  apiBaseUrl: string;
  environmentName: string;
}

export function loadAppConfig(): AppConfig {
  const apiBaseUrl =
    (import.meta.env.VITE_BHASHAVAANI_API_URL as string | undefined) ??
    'http://127.0.0.1:6001';
  const environmentName =
    (import.meta.env.VITE_BHASHAVAANI_ENV as string | undefined) ?? 'local';

  return { apiBaseUrl, environmentName };
}

export function isRemote(config: AppConfig): boolean {
  return config.environmentName !== 'local';
}
